libs/features/mas/web-components/src/merch-datasource.js

```diff
@@ -148,60 +163,92 @@ const cache = new FragmentCache();
  */
 export class MerchDataSource extends HTMLElement {
     /**
-     * @type {import('./aem.js').AEM}
+     * @type {import('@adobe/mas-web-components').AEM}
      */
     #aem;
     cache = cache;
 
+    /**
+     * @type {HtmlElement[]}
+     */
+    refs = [];
+
     /**
      * @type {string} fragment path
      */
     path;
 
+    /**
+     * Consonant styling for CTAs.
+     */
+    consonant = false;
+
+    /**
+     * Internal promise to track the readiness of the web-component to render.
+     */
+    #readyPromise;
+
     static get observedAttributes() {
-        return ['source', 'path'];
+        return ['source', 'path', 'consonant'];
     }
 
     attributeChangedCallback(name, oldValue, newValue) {
         this[name] = newValue;
     }
 
     connectedCallback() {
+        this.consonant = this.hasAttribute('consonant');
+        this.clearRefs();
         const bucket =
-            this.getAttribute(ATTR_AEM_BUCKET) ??
-            document.querySelector('mas-studio')?.getAttribute(ATTR_AEM_BUCKET);
+            this.getAttribute(ATTR_AEM_BUCKET) ?? 'publish-p22655-e59341';
         this.#aem = new AEM(bucket);
-        this.fetchData();
+        this.refresh(true);
     }
 
-    refresh() {
-        this.cache.remove(this.path);
-        this.fetchData();
+    clearRefs() {
+        this.refs.forEach((ref) => {
+            ref.remove();
+        });
+    }
+
+    async refresh(flushCache = true) {
+        if (!this.path) return;
+
+        if (this.#readyPromise) {
+            const ready = await Promise.race([
+                this.#readyPromise,
+                Promise.resolve(false),
+            ]);
+            if (!ready) return; // already fetching data
+        }
+
+        this.clearRefs();
+        this.refs = [];
+        if (flushCache) {
```

Comment:
I see multiple occurrences, but I'll only comment here: the preferable way to write this is `if (something) doSomething();`

Reply:
We need to agree on a consistent format. If (when) somebody else picks up the work, they should write the code in a similar matter.

Reply:
This is MAS engineering preference. w.r.t `if`s,  Milo codebase is not consistent already. If you want to enforce a coding style, please start with a Github discussion. Then each them can address it at their pace.

Reply:
@narcis-radu am i right to understand this is not a requested change? i understand the need for coding rules, but i guess we should probably discuss that in a broader place before making them requested (discussion?)
also i like additional info in the comment to differentiate the two types of comments (minor or requested change), when outcome status is "requested changes"

Reply:
@narcis-radu - just a nit (for now).
@yesil - All code should follow the Milo guidelines. The patterns have been silently enforced so far, but there are some misses since PRs have been approved when this was the only nit (something similar to this PR, right?).
I have created a thread to discuss preferences - https://github.com/orgs/adobecom/discussions/2813